### ISSUE TITLE
Remove tickCount prop

### ISSUE THREAD

**Issue by boygirl** [FormidableLabs/victory-chart#266]
d3 doesnt return the exact number of ticks specified by tickCount, so this prop is misleading. Better to use an array of `tickValues` and get exactly what you want.

**Comment by dandelany** [comment]
FWIW I think this prop is super useful and the underlying d3 method is the method I use/have used in ~90% of the charts I build. "I want 10-ish nicely spaced ticks" (where the interval is more important than the count) is a far more common use case for me than "I want exactly 10 ticks regardless of interval" or "I want exactly these given ticks". It's not misleading if it's documented IMHO.

**Comment by boygirl** [comment]
Fair enough, I'll change this task to improving the documentation of this prop, and adding a prop validation to ensure that values are > 0

**Comment by dandelany** [comment]
:+1: Thanks!

**Comment by stephenotalora** [comment]
Hi @boygirl! 

Actually `tickValues` and `tickCount` share the same behavior, I just didn't notice on friday because my call to generate a ticks range (with d3Array `ticks`) was still generating an array of length > 0.   

however having a tick count of one along the y axis doesn't really allow me to accomplish the styling that I want and we didn't really notice the ticks output because the default JSFiddle `grey` background.

if you attempt to set up a grid with just ticks a long the x-axis victory throws a `TypeError: Cannot read property 'axis' of undefined` an example is available here: https://jsfiddle.net/e5ohe1g0/3/ if you remove the `tickCount` prop on the `dependentAxis` it works, however the grid fallbacks to `defaultProps` and generates a 5x5 grid output which is not quite what we want to accomplish and if you pass an empty array of `tickValues` the same error occurs.

the issue is here: `victory-axis.js line 385` where `baseProps` has an undefined value at the key index. We should check whether baseProps is defined and valid to use before we attempt to use the `axis` property on it.

@boygirl, @ebrillhart I can send a pr but not 100% if there are any implications. Please let me know.

**Comment by stephenotalora** [comment]
@boygirl, @ebrillhart ^ is a small patch. Not quire sure if that is the behavior that you want to go for, but please let me know other wise.
